src/plugins/discover/public/application/components/no_results/no_results.tsx

```diff
@@ -157,33 +184,120 @@ export const DiscoverNoResults = ({ timeFieldName, queryLanguage }: Props) => {
   //   );
   // }
 
+  const [savedQueries, setSavedQueries] = useState<SavedQuery[]>([]);
+
+  useEffect(() => {
+    const fetchSavedQueries = async () => {
+      const { queries: savedQueryItems } = await savedQuery.findSavedQueries('', 1000);
+      setSavedQueries(
+        savedQueryItems.filter((sq) => query?.language === sq.attributes.query.language)
+      );
+    };
+
+    fetchSavedQueries();
+  }, [setSavedQueries, query, savedQuery]);
+
+  const tabs = useMemo(() => {
+    const buildSampleQueryBlock = (sampleTitle: string, sampleQuery: string) => {
+      return (
+        <>
+          <EuiText size="s">{sampleTitle}</EuiText>
+          <EuiSpacer size="s" />
+          <EuiCodeBlock isCopyable>{sampleQuery}</EuiCodeBlock>
+          <EuiSpacer size="s" />
+        </>
+      );
+    };
+
+    const sampleQueries = [];
+    if (query?.dataset?.type && datasetService.getType(query.dataset.type)?.getSampleQueries) {
+      sampleQueries.push(
+        ...datasetService.getType(query.dataset.type)!.getSampleQueries!(
+          query.dataset,
+          query.language
+        )
+      );
+    }
+    if (query && languageService.getLanguage(query?.language)?.sampleQueries) {
+      sampleQueries.push(...(languageService.getLanguage(query.language)!.sampleQueries ?? []));
+    }
+
+    return [
+      ...(sampleQueries.length > 0
+        ? [
+            {
+              id: 'sample_queries',
+              name: i18n.translate('discover.emptyPrompt.sampleQueries.title', {
+                defaultMessage: 'Sample Queries',
+              }),
+              content: (
+                <Fragment>
+                  <EuiSpacer size="s" />
+                  {sampleQueries
+                    .slice(0, 5)
+                    .map((sampleQuery) =>
+                      buildSampleQueryBlock(sampleQuery.title, sampleQuery.query)
+                    )}
+                </Fragment>
+              ),
+            },
+          ]
+        : []),
+      ...(savedQueries.length > 0
```

Comment:
ah even more so now i think we should consider unifying saved query and sample query at least since they both have the same content